tests/framework/callbacks/test_notification.py

```diff
@@ -96,22 +100,56 @@ def test_givenANotificationCallbackWithExperimentName_whenTrainingLoop_thenSendN
             callbacks=[notification_callback],
         )
 
-        self._test_notificator_call(logs, experiment_name=a_experiment_name)
+        self._build_notificator_call(logs, experiment_name=a_experiment_name)
+
+        self.assert_mock_calls(self.call_list)
+
+    def test_givenANotificationCallback_whenTestLoop_thenSendNotification(self):
+        notification_callback = NotificationCallback(notificator=self.notificator_mock)
+        res = self.model.evaluate(
+            x=self.some_x_data, y=self.some_y_data, callbacks=[notification_callback], return_dict_format=True
+        )
+
+        self._build_notificator_call(res, mode="testing")
 
-    def _test_notificator_call(self, logs: Dict, experiment_name=None):
+        self.assert_mock_calls(self.call_list)
+
+    def test_givenANotificationCallbackWithExperimentName_whenTestLoop_thenSendNotificationWithExperimentName(self):
+        a_experiment_name = "A experiment name"
+        notification_callback = NotificationCallback(
+            notificator=self.notificator_mock, experiment_name=a_experiment_name
+        )
+        res = self.model.evaluate(
+            x=self.some_x_data, y=self.some_y_data, callbacks=[notification_callback], return_dict_format=True
+        )
+
+        self._build_notificator_call(res, mode="testing", experiment_name=a_experiment_name)
+
+        self.assert_mock_calls(self.call_list)
+
+    def assert_mock_calls(self, call_list: List):
+        method_calls = self.notificator_mock.method_calls
+        self.assertEqual(len(method_calls), len(call_list))
+        self.assertEqual(method_calls, call_list)
+
+    def _build_notificator_call(self, logs, experiment_name=None, mode: str = "training"):
```

Comment:
Should return the call_list instead of putting it as an attribute.

Reply:
Since the method doesn't use the self, do you prefer a static method of a function?

Reply:
97ba9f6